codeclimate-validate-config/bin/run

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env ruby
+#
+# TODO: validate engine names as valid engines. this requires access to the
+# shared engine registry, so we should think carefully if that functionality is
+# worth the coupling.
```

Comment:
Is the intention of `codeclimate engines:list` to list all *available* engines? If so, we could use that here to perform this validation. That would be well in line with the spirit of this architecture. The `codeclimate-engines` subcommand could encapsulate access to the registry.